blake2f/src/f.rs

```diff
@@ -0,0 +1,129 @@
+/// Implementation of blake2 compression function F.
+///
+/// This was copied from https://gist.github.com/seunlanlege/fa848401d316c52919f6e554fba6870b
+/// with some modifications. It was initially written by Seun Lanlege and has no explicit license.
```

Comment:
Oh only noticed this now. Could you ask him on the gist to license it under Apache-2.0 or MIT or something like that?